### PR TITLE
Move to ICU BreakerIterator in TextShaper

### PR DESCRIPTION
**Pull request by Fahad-Alsaidi**
LineBreaking now works fine with CJK & Thai but doesn't work with Latin or Arabic.

### CODE REVIEW

File: scribus/text/textshaper.cpp [hunk 1]
```diff
@@ -14,6 +13,7 @@
 #include "storytext.h"
 #include "styles/paragraphstyle.h"
 
+BreakIterator* TextShaper::m_lineIterator = NULL;
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: Move this to story text like the other iterators, make `getLineIterator()` static and call it from there.

File: scribus/text/textshaper.cpp [hunk 2]
```diff
@@ -396,8 +395,9 @@ QList<GlyphCluster> TextShaper::shape()
 			if (textRun.dir == UBIDI_RTL)
 				run.setFlag(ScLayout_RightToLeft);
 
-			lineBoundery.setPosition(firstCluster);
-			if (lineBoundery.isAtBoundary())
+			BreakIterator* lineBoundery = getLineIterator();
+			assert(lineBoundery != NULL);
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: I don’t like this assert, we should either do nothing the rare case that and iterator can’t be created or just use some simplistic fallback e.g. the old Scribus line breaking stuff, but a hard crash is not a solution.

Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: Asserts generally are reserved for things that never expected to fail and we have code that assume they never fail and will crash anyway, so we crash early to make the source of the failure clear.

File: scribus/text/textshaper.cpp [hunk 3]
```diff
@@ -396,8 +395,9 @@ QList<GlyphCluster> TextShaper::shape()
...
+			if (lineBoundery->isBoundary(firstCluster))
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: You don’t call `setText()` anywhere!

File: scribus/text/textshaper.cpp [hunk 4]
```diff
@@ -396,8 +395,9 @@ QList<GlyphCluster> TextShaper::shape()
 			if (textRun.dir == UBIDI_RTL)
 				run.setFlag(ScLayout_RightToLeft);
 
-			lineBoundery.setPosition(firstCluster);
-			if (lineBoundery.isAtBoundary())
+			BreakIterator* lineBoundery = getLineIterator();
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: You should call this once before looping over the runs.

**Comment by Fahad-Alsaidi** [comment]
Updated but still not working as expected. Any help are welcome.

File: scribus/text/textshaper.h [hunk 5]
```diff
@@ -5,6 +5,7 @@
 #include <QMap>
 #include <QString>
 #include <unicode/uscript.h>
+#include <unicode/brkiter.h>
```
Comment (khaledhosny) [inline scribus/text/textshaper.h]: We don’t need this include.

File: scribus/text/textshaper.cpp [hunk 6]
```diff
@@ -14,7 +13,6 @@
 #include "storytext.h"
 #include "styles/paragraphstyle.h"
 
-
 TextShaper::TextShaper(PageItem *item, StoryText &story, int first, bool singlePar)
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: This cosmetic change is unnecessary.

**Pull request merged**